src/ray/rpc/gcs_server/gcs_rpc_server.h

```diff
@@ -335,6 +342,43 @@ class MonitorGrpcService : public GrpcService {
   MonitorGcsServiceHandler &service_handler_;
 };
 
+// Legacy RPC interface for supporting autoscaler v1
+class LegacyAutoscalerGcsServiceHandler {
```

Comment:
could we just piggyback on the existing autoscaler state service for the `HandleGetAllResourceUsage`?

Reply:
It's possible, we would just have to make new message types for `GetAllResourceUsageRequest` and `GetAllResourceUsageReply`, marshal them to python, and also regenerate python protos. We could also do it by having `autoscaler.proto` take a dependency on `gcs.proto` which feels hacky (because the "temporary" dependency could grow over time if we're not careful) but is a smaller change. 

Adding a small service handler just minimizes mixing new autoscaler with old autoscaler by creating a concrete piece of "obvious tech debt" that can be pretty cleanly removed. The size of the code change compared to doing it in `autoscaler.proto` would be around the same, but we wouldn't have the overhead of another service handler.